core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1255,15 +1257,19 @@ class ReplicaManager(val config: KafkaConfig,
         val partitionsToBeFollower = partitionStates -- partitionsTobeLeader.keys
 
         val highWatermarkCheckpoints = new LazyOffsetCheckpoints(this.highWatermarkCheckpoints)
-        val partitionsBecomeLeader = if (partitionsTobeLeader.nonEmpty)
+        val partitionsBecomeLeader = if (partitionsTobeLeader.nonEmpty) {
+          stateChangeLogger.info(s"Handling LeaderAndIsr request correlationId $correlationId from controller $controllerId " +
```

Comment:
I was thinking of logging the partition count at the beginning of becomeLeaderOrFollower(). Otherwise, it's kind of weird to see the request logging after the logging for "Ignoring".